opencog/spacetime/pointmemory/PointMemorySCM.cc

```diff
@@ -47,13 +51,16 @@ class PointMemorySCM
 	// Check if the named map exists.
 	void have_map(const Handle&);
 
-	std::map<Handle, TimeOctomap<Handle>*> tsa;
+	std::map<Handle, std::shared_ptr<TimeOctomap<Handle>>> tsa;
 	time_pt get_map_time(const Handle& map_name, const Handle& elapse);
 
 public:
+    static std::map<Handle,Handle> handle_octomap_map;
```

Comment:
From what I can tell, you don't actually need this. If you already have the OctoMap Node, you don't need to look it up a second time!  In the code below, `map_name` is already the map that you want.  The indirection is not needed ...